real_check/real_check.ps1

```diff
@@ -0,0 +1,57 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+function Check-Symbols {
+    param(
+        [string] $lib
+    )   
+        try {
+            [string]$output = cmd.exe /c dumpbin /ALL $lib
+        }
+        catch {
+            Write-Error  "Unable to run dumpbin on given lib"
+            return 2
+        }
+        $start_token = "public symbols"
+        $end_token = "Archive member name"
+        try {
+            [int] $start = $output.IndexOf($start_token)
+            [int] $end = $output.IndexOf($end_token, $start)
+            $symbol_text = ($output.Substring($start + $start_token.Length+1, $end-$start-$start_token.Length-1))
+            $split_symbols = $symbol_text.Split('',[StringSplitOptions]::RemoveEmptyEntries)
+        }
+        catch {
+            Write-Error  "Unable to parse dumpbin output"
+            return 2
+        }
+        if($split_symbols.Length -lt 2) {
+            Write-Error  "Unable to parse dumpbin output"
```

Comment:
I would love to have some comments with an example of what this does.
Like:
"dumpbin gives us this format: 2 columns containing symbol name, etc. So we store that output in a hashset, etc."

Just so that it is eaasier for you (and the others) to modify this if needed when we come back to it in a year.

Reply:
Agreed, adding comments in next iteration.

Reply:
I love that you added the comment with the examples of using the .ps1.
What I meant though is let's add a comment here at line 48 (approx), stating why we are writing the code as such.

Think about it like this:
In 6 months if you come accross this will you remember what the output format of dumpbin is? Wouldn't it be nice to have a glimpse of why you chose to process every other entry?